pkg/kubectl/apply/strategy/merge_visitor.go

```diff
@@ -41,7 +41,10 @@ func (v mergeStrategy) MergeList(e apply.ListElement) (apply.Result, error) {
 	if result, done := v.doAddOrDelete(e); done {
 		return result, nil
 	}
-
```

Comment:
I am not sure about this, but why not doing this before `doAddOrDelete`. Same for others. also it might be a good idea to change it to `doConflictDetect`, I assume it's internal.

Reply:
I believe that doAddOrDelete() has no merge logic if adding or deleting. It should go prior to merge. If "not done" returns from doAddOrDelete(), it starts to merge and conflict detection if optioned.